internal/policy/root.go

```diff
@@ -27,35 +29,49 @@ func InitializeRootMetadata(key *tuf.Key) *tuf.RootMetadata {
 	return rootMetadata
 }
 
-// AddTargetsKey adds targetsKey as a trusted public key in rootMetadata for the
+// AddTargetsKey function is used to add the 'targetsKey' as a trusted public key in 'rootMetadata' for the
 // top level Targets role.
-func AddTargetsKey(rootMetadata *tuf.RootMetadata, targetsKey *tuf.Key) *tuf.RootMetadata {
+func AddTargetsKey(rootMetadata *tuf.RootMetadata, targetsKey *tuf.Key) (*tuf.RootMetadata, error) {
+	if rootMetadata == nil {
```

Comment:
Hm, the only consumer for this is `repository.AddTopLevelTargetsKey`, which is after the current state is loaded (during which the state is verified as well). I'm ambivalent on adding this check.

Reply:
>I'm ambivalent on adding this check.

Why? This will avoid panics.

Reply:
Agreed, it'll avoid panics on direct invocations, but my point was we don't really expect that for this package. That said, redundant checks are probably better, so let's proceed. :)